src/clj/ataru/applications/application_store.clj

```diff
@@ -617,23 +617,51 @@
-                                (let [hakukohde-event {:application_key  application-key
-                                                       :event_type       "hakukohde-review-state-change"
-                                                       :new_review_state (:state review-to-store)
-                                                       :review_key       hakukohde-review-requirement
-                                                       :hakukohde        (:hakukohde review-to-store)
-                                                       :virkailija_oid   (-> session :identity :oid)}]
-                                  (store-and-log-review-event connection hakukohde-event session))))))
+    (let [connection                  {:connection conn}
+          review-to-store             {:application_key application-key
+                                       :requirement     hakukohde-review-requirement
+                                       :state           hakukohde-review-state
+                                       :hakukohde       hakukohde-oid}
+          existing-duplicate-review   (yesql-get-existing-application-hakukohde-review review-to-store connection)
+          existing-requirement-review (first (yesql-get-existing-requirement-review review-to-store connection))]
+      (when (empty? existing-duplicate-review)
+            (yesql-upsert-application-hakukohde-review! review-to-store connection)
+            (when session
+                  (auditlog-review-modify review-to-store existing-requirement-review session))
+            (let [hakukohde-event {:application_key  application-key
+                                   :event_type       "hakukohde-review-state-change"
+                                   :new_review_state (:state review-to-store)
+                                   :review_key       hakukohde-review-requirement
+                                   :hakukohde        (:hakukohde review-to-store)
+                                   :virkailija_oid   (-> session :identity :oid)}]
+              (store-and-log-review-event connection hakukohde-event session))))))
+
+(defn save-automatic-application-hakukohde-review
+  [application-key hakukohde-oid hakukohde-review-requirement hakukohde-review-state session]
+  (jdbc/with-db-transaction [conn {:datasource (db/get-datasource :db)}]
+    (let [connection                  {:connection conn}
+          review-to-store             {:application_key application-key
+                                       :requirement     hakukohde-review-requirement
+                                       :state           hakukohde-review-state
+                                       :hakukohde       hakukohde-oid}
+          existing-modified-by-user?  (->> (yesql-get-application-events {:application_key application-key} connection)
+                                           (sort-by :time)
```

Comment:
Tähä viel ekaks `(filter #(and (= hakukohde-oid (:hakukohde %)) (= hakukohde-review-requirement (:requirement %))))`.